File: 2D-World-Simulator-Java/OPP-Java/src/op/JavaProject/Organisms/Plants/SowThistle.java
package op.JavaProject.Organisms.Plants;

import op.JavaProject.Functionality.BoardPoint;
import op.JavaProject.Functionality.Species;
import op.JavaProject.Game.Plant;
import op.JavaProject.Game.World;

import java.awt.*;

import static op.JavaProject.Game.OrganismsStatics.MAX_ATTEMPT;

public class SowThistle extends Plant {

    public SowThistle(World organismWorld, BoardPoint point) {
        super(organismWorld, point);
        this.strength = 0;
        this.initiative = 0;
        this.organismSign = "*";
        this.organismSpecie = Species.SOW_THISTLE;
        this.organismColor = new Color(254, 255, 0);
    }

    //sowThistle has 3 attempts to sow in one round
    @Override
    public void Action() {
        int attemps = MAX_ATTEMPT;
        while (attemps-- > 0) {
            super.Action();
        }

    }

}

File: 2D-World-Simulator-Python/OPP-Python/Organisms/Plants/SowThistle.py
from Game.OrganismStatics import *
from Game.Plant import Plant


class SowThistle(Plant):
    def __init__(self, organism_world, point):
        super().__init__(organism_world, point, 0, 0, "*", OrganismSpecie.SOW_THISTLE, "#feff00")

    # sowThistle has 3 attempts to sow in one round
    def action(self):
        attempts = MAX_ATTEMPT
        while attempts > 0:
            super().action()
            attempts -= 1
